frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -1,18 +1,202 @@
-import React from "react";
-import { GridContainer } from "@trussworks/react-uswds";
+import React, { ReactNode, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import {
+    GridContainer,
+    SiteAlert,
+    Icon,
+    Form,
+    FormGroup,
+    Dropdown,
+    Button,
+} from "@trussworks/react-uswds";
 
 import { AuthElement } from "../AuthElement";
 import { withCatchAndSuspense } from "../RSErrorBoundary";
+import { showError } from "../AlertNotifications";
+import Spinner from "../Spinner";
+import { USLink } from "../USLink";
 import { MemberType } from "../../hooks/UseOktaMemberships";
+import { UseOrganizationSenders } from "../../hooks/UseOrganizationSenders";
+import { RSSender } from "../../config/endpoints/settings";
+import useManagePublicKey, {
+    ManagePublicKeyActionType,
+} from "../../hooks/network/ManagePublicKey/ManagePublicKeyHooks";
+import { ContentType } from "../../utils/TemporarySettingsAPITypes";
+
+import { ManagePublicKeyUpload } from "./ManagePublicKeyUpload";
+
+const LightbulbIcon = Icon.Lightbulb;
+
+// TODO: create common component???
+const ManagePublicKeySpinner = ({ message }: { message: ReactNode }) => (
```

Comment:
Wouldn't hurt to break this out into its own module. I have a ticket to replace our usage of the third-party library spinner with our own (based off the spinner svg in one of USWD's modules) so having a local spinner module in place and in-use would help with the eventual swap. A local Spinner component that can take custom className and an optional message prop.

Reply:
Yeah absolutely agree -- no reason not to have a single, generalized component for all spinners!

Reply:
Will do. Validator can also use this common component.